app/src/main/java/com/github/siela1915/bootcamp/HomePageFragment.java

```diff
@@ -88,36 +114,212 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         if(data != null){
             cuis+= data.getString("selected cuisine");
         }
-        //homeTextView.setText(cuis);
-        //homeTextView.setTextSize(30);
+        TextView underlinedText = view.findViewById(R.id.moreFilterTextView);
+        underlinedText.setPaintFlags(underlinedText.getPaintFlags() | Paint.UNDERLINE_TEXT_FLAG);
+        getActivity().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN);
         return view;
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
-        Button button = view.findViewById(R.id.homeFragButton);
+        //Button button = view.findViewById(R.id.homeFragButton);
         recipeListRecyclerView= view.findViewById(R.id.rand_recipe_recyclerView);
         recipeListRecyclerView.setHasFixedSize(true);
         recipeListRecyclerView.setLayoutManager(new GridLayoutManager(getContext(),2));
+        moreFilter= view.findViewById(R.id.moreFilterTextView);
+        filterLayout = view.findViewById(R.id.filter);
+        recipeListLinearLayout = view.findViewById(R.id.recipeListLinearLayout);
+        searchView= view.findViewById(R.id.searchView);
+        searchView.clearFocus();
+        dietBtn = view.findViewById(R.id.btnDiet);
+        allergyBtn=view.findViewById(R.id.btnAllergy);
+        cuisineBtn = view.findViewById(R.id.btnCuisine);
+        //prepTimeBtn= view.findViewById(R.id.btnPrpTime);
+
+
+        cuisineBtn.setOnClickListener(v -> {
+            String [] cuisineTypes= CuisineType.getAll();
+            boolean[] checksum= new boolean[cuisineTypes.length];
+            String title = "Choose your preferred cuisine";
+            popUpDialogBuilder(cuisineTypes,checksum,title,selectedCuisine,CUISINE);
+        });
```

Comment:
Click listeners for cuisine, diet and allergy buttons seem to have similar structures. To avoid code duplication, you might consider extracting the common parts into a separate method.